js/mcqView.js

```diff
@@ -1,175 +1,86 @@
+    if (!this.model.isInteractive()) return;
+
+    const index = $(event.currentTarget).data('adapt-index');
+    const itemModel = this.model.getItem(index);
+    let shouldSelect = !itemModel.get("_isActive");
+
+    if (this.model.isSingleSelect()) {
+      // Assume a click is always a selection
+      shouldSelect = true;
       this.model.resetActiveItems();
-      this.model.resetItems();
-    },
-
-    showCorrectAnswer: function() {
-      this.isCorrectAnswerShown = true;
-      this.update();
-    },
-
-    hideCorrectAnswer: function() {
-      this.isCorrectAnswerShown = false;
-      this.update();
-    },
-
-    update: function() {
-      this.updateSelection();
-      this.updateMarking();
-    },
-
-    updateSelection: function() {
-
-      var isEnabled = this.model.get("_isEnabled");
-
-      this.model.getChildren().each(function(itemModel) {
-
-        var isSelected = this.isCorrectAnswerShown ?
-            itemModel.get("_shouldBeSelected") :
-            itemModel.get("_isActive");
-
-        var index = itemModel.get('_index');
-        this.$('.js-item-label').filter('[data-adapt-index="' + index + '"]')
-            .toggleClass('is-selected', isSelected)
-            .toggleClass('is-disabled', !isEnabled);
-
-        this.$('.js-item-input').filter('[data-adapt-index="' + index + '"]')
-            .prop('checked', isSelected)
-            .prop('disabled', !isEnabled);
-
-      }.bind(this));
-
-    },
-
-    updateMarking: function() {
-
-      var isInteractive = this.model.isInteractive();
-      var canShowMarking = this.model.get('_canShowMarking');
-      var ariaLabels = Adapt.course.get('_globals')._accessibility._ariaLabels;
-
-      this.model.getChildren().each(function(itemModel) {
-
-        var index = itemModel.get('_index');
-        var $itemInput = this.$('.js-item-input').filter('[data-adapt-index="' + index + '"]');
-        var $item = $itemInput.parents('.js-mcq-item');
-
-        if (isInteractive || !canShowMarking) {
-          // Remove item marking
-          $item.removeClass('is-correct is-incorrect');
-          $itemInput.attr('aria-label', $.a11y_normalize(itemModel.get("text")));
-          return;
-        }
+    } else if (shouldSelect && this.model.isAtActiveLimit()) {
+      // At the selection limit, deselect the last item
+      this.model.getLastActiveItem().toggleActive(false);
+    }
 
-        // Mark item
-        var shouldBeSelected = itemModel.get("_shouldBeSelected");
-        var isCorrect = Boolean(itemModel.get("_isCorrect"));
-        var isActive = Boolean(itemModel.get("_isActive"));
+    // Select or deselect accordingly
+    itemModel.toggleActive(shouldSelect);
+  }
 
-        $item
-            .toggleClass('is-correct', isCorrect)
-            .toggleClass('is-incorrect', !isCorrect);
+  // Blank method to add functionality for when the user cannot submit
+  // Could be used for a popup or explanation dialog/hint
+  onCannotSubmit() {}
```

Comment:
I'm not sure this will be needed following the release of FW v5.7.0

Reply:
Removed